scripts/prepare_fsd50k.py

```diff
@@ -0,0 +1,159 @@
+                metadata_path="../dcase_datagen/metadata/fsd50k"):
+        self.dataset_name = dataset_name
+        self.dataset_home = dataset_home
+        self.download_dataset = download_dataset
+        self.music = music_dataset
+        self.music_home = music_home
+        self.download_music = download_music
+        self.dcase_sound_events_txt = dcase_sound_events_txt
+        self.metadata_path = metadata_path
+        self.url_fsd_selected_txt = "https://zenodo.org/record/6406873/files/FSD50K_selected.txt"
+        self.fsd_to_dcase_dict = {} # Dict containing FSD50K to DCASE file mappings
+
+    def download_dataset(self):
+        self.fsd50k.download()
+
+    def load_dataset(self):
+        assert self.dataset_home is not None or not self.download \
+        , "Dataset home path must be provided when download is enabled."
+        self.fsd50k = soundata.initialize(self.dataset_name, self.dataset_home)
+        if self.download_dataset:
+            self.download_dataset() # download fsd50k
+        if self.music == "fma_small" and self.download_music:
+            self.music = FMADataLoad(dataset_home=self.music_home) # download fma_small
+            self.music.load_dataset()
+            self.music.gen_dataset_splits()
+            self.music.save_fma_to_dcase_json()
+        self.gen_fsd_to_dcase_dict()
+
+    def save_fsd_to_dcase_json(self):
+        with open(os.path.join(self.metadata_path, 'fsd_to_dcase.json'), 'w') as json_file:
+            json.dump(self.fsd_to_dcase_dict, json_file, indent=4)
+
+    def get_filenames(self, fold="train"):
+        if fold not in self.fsd_to_dcase_dict.keys():
+            raise ValueError("Invalid fold argument. The fold must be 'train' or 'test'.")
+        return self.fsd_to_dcase_dict[fold]
+
+    def gen_fsd_to_dcase_dict(self):
+        if not os.path.exists(self.dcase_sound_events_txt):
+            download_file(self.url_fsd_selected_txt, self.dcase_sound_events_txt)
+
+        with open(self.dcase_sound_events_txt, 'r') as file:
+            for line in file:
+                dcase_path = line.strip()
+                parts = line.split('/')
+                track_id = os.path.basename(dcase_path)
+                super_class = parts[0]
+                fold = parts[1]
+                class_name = parts[2]
+                dcase_path = f"{super_class}/{fold}/{class_name}/{track_id}"
+                fsd50k_fold = 'FSD50K.dev_audio' if fold == 'train' else 'FSD50K.eval_audio'
+                if fold not in self.fsd_to_dcase_dict:
+                    self.fsd_to_dcase_dict[fold] = {}
+                self.fsd_to_dcase_dict[fold][dcase_path] = os.path.join(self.dataset_home, f"{fsd50k_fold}/{track_id}")
+        self.fsd_to_dcase_dict['train'].update(self.music.fma_to_dcase_dict['train'])
+        self.fsd_to_dcase_dict['test'].update(self.music.fma_to_dcase_dict['test'])
+
+if __name__ == "__main__":
```

Comment:
This is very neat. Thanks.

It would be good if the user can easily pass parameters via the command line. Perhaps consider using [fire](https://github.com/google/python-fire)

Reply:
I decided not to use fire. I believe using fire adds another layer of user dependent complications. Instead I think we should follow a fixed yaml config file. Let me know what you think. I do believe we are adding many yamls in different parts of this repo though